addons/project/wizard/project_share_wizard.py

```diff
@@ -35,14 +35,38 @@ def _compute_resource_ref(self):
             else:
                 wizard.resource_ref = None
 
+    def action_send_mail_post_warning(self):
+        self.ensure_one()
+        portal_partners = self.partner_ids.filtered('user_ids')
+        note = self._get_note()
+        non_portal_partners = self.partner_ids - portal_partners
+
+        self.resource_ref._add_collaborators(self.partner_ids)
+        self._send_public_link(note, portal_partners)
+        self._send_signup_link(note, partners=non_portal_partners)
+        self.resource_ref.message_subscribe(partner_ids=self.partner_ids.ids)
+        return {'type': 'ir.actions.act_window_close'}
+
     def action_send_mail(self):
         self.ensure_one()
         if self.access_mode == 'edit':
             portal_partners = self.partner_ids.filtered('user_ids')
-            note = self._get_note()
-            self.resource_ref._add_collaborators(self.partner_ids)
-            self._send_public_link(note, portal_partners)
-            self._send_signup_link(note, partners=self.partner_ids - portal_partners)
-            self.resource_ref.message_subscribe(partner_ids=self.partner_ids.ids)
-            return {'type': 'ir.actions.act_window_close'}
+            non_portal_partners = self.partner_ids - portal_partners
+
+            if len(non_portal_partners) > 0:
+                warning = self.env['project.share.warning.wizard'].create({
+                    'project_share_id': self.id,
+                    'non_portal_partner_ids': non_portal_partners
```

Comment:
In general, we prefer adding a comma at the end even if it the final item passed in the dictionary to avoid changing your line when we want to add an additional item in this dictionary. 
(Same comment for the items added in a list).
🙂 
```suggestion
                    'non_portal_partner_ids': non_portal_partners,
```